### DISCUSSION TITLE
Issue with editing source code

### DISCUSSION THREAD

**Discussion by NivedithaCV** [DOI-USGS/ISIS3#5296] [Q&A] [answered]
**Issue Description**

I have encountered a problem while editing the source code. Whenever I make changes and compile the code I encounters the same error. However, when I reinstall the virtual environment created for the project, the code runs without any problems.

**Steps to Reproduce**
1. I followed the instructions in [link to ISIS3 setup documentation](https://github.com/DOI-USGS/ISIS3/wiki/Developing-ISIS3-with-cmake#getting-started-with-github) to build ISIS in a virtual environment named "dev," using the `make` command.
2. I edited the code in `ISIS3/isis/src/apollo/apps/apollo2isis/main.cpp`.
3. I attempted to build the app using the command `make install apollo2isis_app`.
4. The compilation fails. I remove the env `conda remove -n dev --all command`
5. Creates the env again `conda env create -n dev -f environment.yml`
6. Builds the apollo2isis_app successfully

**Actual Behavior**
- Step 1: Successfully built.
- Compiling the edited source code produced the following error:

```cpp
/home/niser/anaconda3/envs/dev/version:1:1: error: too many decimal points in number
    1 | 7.2.0 # Public version number
      | ^~~~~
/home/niser/anaconda3/envs/dev/version:1:7: error: stray '#' in program
    1 | 7.2.0 # Public version number
      |       ^
/home/niser/anaconda3/envs/dev/version:2:6: error: invalid digit "9" in octal constant
    2 | 2023-09-19 # Release date
      |      ^~
/home/niser/anaconda3/envs/dev/version:2:12: error: stray '#' in program
    2 | 2023-09-19 # Release date
      |            ^
/home/niser/anaconda3/envs/dev/version:3:6: error: stray '#' in program
    3 | beta # Release stage (alpha, beta, stable)
      |      ^
...
/home/niser/anaconda3/envs/dev/include/nlohmann/detail/iterators/iter_impl.hpp:479:27: error: 'class nlohmann::json_abi_v3_11_2::basic_json<>' has no member named 'm_type'
  479 |         switch (m_object->m_type)
      |                 ~~~~~~~~~~^~~~~~
make[2]: *** [objects/CMakeFiles/apollo2isis_app.dir/build.make:76: objects/CMakeFiles/apollo2isis_app.dir/apollo/apps/apollo2isis/main.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:19990: objects/CMakeFiles/apollo2isis_app.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

- After removing the "dev" virtual conda environment and creating a new environment named "dev" with the packages listed in the "environment.yml" file in the ISIS3 folder, the app built and ran without any errors.

**Environment information**

1. conda version: conda 23.7.3
2. Operating system: Ubuntu 20.04.6 LTS

**Additional Context**
It doesn't matter which file I edit; I have tried "rolo2isis" as well, which gave me the same error. After reinstalling the environment, the code ran without any errors. I can reproduce this issue consistently. The error shared here only shows a small part.

If anyone has experienced this issue or can point out any steps I am doing wrong, please help.

Thank you for your assistance in resolving this issue!

**Comment by AustinSanders** [comment, marked as answer]
Hey @NivedithaCV, I thought that this problem was being tracked in an issue, but I'm having trouble finding it. I'll try to search around a bit more and potentially link to that issue if I can find anything.

My understanding is that this is an issue in which the 'version' file is misidentified as a source file, and our build system attempts to compile it.  If I remember correctly, I've gotten around this issue by deleting the version file, which should be as simple as `rm /home/niser/anaconda3/envs/dev/version`.

**Reply by NivedithaCV** [reply]
Thank you, that solved the issue. Is it possible to prevent the build from generating the version file inside the environment? Currently, I need to delete the version file in every build session.

**Reply by AustinSanders** [reply]
I haven't tested it, but it looks like [this line](https://github.com/DOI-USGS/ISIS3/blob/dev/isis/CMakeLists.txt#L420) may be responsible for the creation of the version file.  Commenting out this line may prevent the file from being generated.

**Reply by NivedithaCV** [reply]
Thank you for the suggestion, but unfortunately it didn't work.
